### DISCUSSION TITLE
Customizable Model Selection for Improved User Experience

### DISCUSSION THREAD

**Discussion by FedeCuci** [open-webui/open-webui#5902] [General]
**Is your feature request related to a problem? Please describe.**
Different companies offer many models through their API. OpenAI for example, currently offers 20 models. When adding the OpenAI API key in Open Web UI, the dropdown menu for choosing which model to use always shows all models, which is too verbose. I already know that I want to use just a handful of them, so it is frustrating to always have to scroll through all of them.

**Describe the solution you'd like**
Although there are different solutions to this problem, I think there could be a place in settings where the user can choose which models are available in the dropdown menu. Possibly this could be a dropdown menu in the interface settings tab, just under the "Default Model" menu. The user can then click on the models they would like to have in the drop-down menu in the chat interface. I think this would enhance the user experience.

**Comment by silentoplayz** [comment]
You can hide models from the model dropdown selector in the `Workspace` -> `Models` section.
Personally, I hide my Ollama embedding models from the model dropdown selector:
![image](https://github.com/user-attachments/assets/723cdf41-a433-496c-9343-40330c99fe04)

![image](https://github.com/user-attachments/assets/cdda827b-01f9-4d1a-b780-8169e64b4643)

**Comment by FedeCuci** [comment]
That's indeed what I was looking for, thank you! I thought this feature would be found in the settings, but this works too. Do you think adding an opt-in model selection would make sense? For instance, I'm using an API key from OpenRouter with 70+ models, but I only want to display about 10. It seems more intuitive to choose which ones to show rather than exclude the ones you don't want to show. I realize it's a small feature, but it could be handy for many users.

**Comment by FedeCuci** [comment]
Another way to achieve this could be to allow users to select multiple models at once and perform the same action on all of them using checkboxes.

**Reply by justinh-rahb** [reply]
You can hold shift to make the "Hide" button visible on each row and spam-click them.

**Reply by gakugaku** [reply]
> You can hold shift to make the "Hide" button visible on each row and spam-click them.

Very useful comment, but the operation is tricky so I can't notice the function.😅

**Comment by silenceroom** [comment]
this function seems only works on the models from the "admin panel". The custom models in "workspace" can't be hided. :(
